package/MDAnalysis/lib/util.py

```diff
@@ -1267,6 +1267,54 @@ def fixedwidth_bins(delta, xmin, xmax):
     dx = 0.5 * (N * _delta - _length)  # add half of the excess to each end
     return {'Nbins': N, 'delta': _delta, 'min': _xmin - dx, 'max': _xmax + dx}
 
+def get_weights(atoms, weights):
+    """Check that a `weights` argument is compatible with `atoms`.
+
+    Parameters
+    ----------
+    atoms : AtomGroup or array_like
+        The atoms that the `weights` should be applied to. Typically this
+        is a :class:`AtomGroup` but because only the length is compared,
+        any sequence for which ``len(atoms)`` is defined is acceptable.
+    weights : {"mass", None} or array_like
+        All MDAnalysis functions or classes understand "mass" and will then
+        use ``atoms.masses``. ``None`` indicates equal weights for all atoms.
+        Using an ``array_like`` assigns a custom weight to each element of
+        `atoms`.
+
+    Returns
+    -------
+    weights : array_like or None
```

Comment:
The reason why I kept the special case `None` is so that this can be directly fed to
```
np.average(positions, weights=weights)
```

`None` saves the multiplication of the `weights` with the positions, broadcasting and what not. It's a completely different code path in `np.average()`. Furthermore, it also saves us time in `rotation_matrix()` and friends. It might not make a big difference for a few coordinates but for large systems I would want to avoid *any* array/array broadcast/multiplication.

I think this specific case is sufficiently well motivated to be kept. Or what is your argument for returning a `np.ones(len(atoms))`?